### PR TITLE
feat: upgrade to construct models on first call instead of on controller instanciation

### PR DESCRIPTION
**Pull request by phpsa**

### CODE REVIEW

File: src/Http/Api/Contracts/HasModel.php [hunk 1]
```diff
@@ -78,7 +82,7 @@ public function getPostmanModel(): string
     protected function unguardIfNeeded()
     {
         if ($this->unguard) {
-            self::$model->unguard();
+            $this->getModel()::unguard();
```
Comment (samatcd) [inline src/Http/Api/Contracts/HasModel.php:85]: Is static call correct? Previous was not static? Based on existing call this should be:

```
$this->getModel()->unguard();
```

**Review by samatcd — CHANGES_REQUESTED** [review]
Looks good!

File: src/Http/Api/Contracts/HasModel.php (hunk 1 above)
Comment (samatcd) [reply src/Http/Api/Contracts/HasModel.php:85]: It does look like it should be static, but curious as to why it was using `->` previously.

Comment (phpsa) [reply src/Http/Api/Contracts/HasModel.php:85]: This should have always been static so would suggest keeping new call

**Pull request merged**